src/main/java/net/helpscout/api/Parser.java

```diff
@@ -59,11 +57,10 @@ public Customer getCustomer(String json) {
 
 	public <T> T getObject(JsonElement item, Class<T> clazzType) {
 		JsonThreadLocal.set(item);
-
-		T clazz = builder.create().fromJson(item, clazzType);
-
-		JsonThreadLocal.unset();
-
-		return clazz;
+		try {
+			return builder.create().fromJson(item, clazzType);
+		} finally {
+			JsonThreadLocal.unset();
```

Comment:
It's necessary right now to share state between deserializers. See for instance [here](https://github.com/helpscout/helpscout-api-java/blob/develop/src/main/java/net/helpscout/api/adapters/PersonRefAdapter.java#L21). The GSON user guide [advises using thread locals](https://sites.google.com/site/gson/gson-user-guide#TOC-Sharing-State-Across-Custom-Serializers-and-Deserializers) as one way of sharing state in a thread-safe way.

FWIW, I think it's pretty ugly as well, but had to draw the line somewhere. I'd like to revisit removing it completely in v 2.0 of this library.